ffi/src/x509.rs

```diff
@@ -215,4 +215,188 @@ pub mod ffi {
             Ok(Box::new(Cert(cert)))
         }
     }
+
+    #[diplomat::opaque]
+    pub struct CertVec(pub Vec<certificate::Cert>);
+
+    impl CertVec {
+        pub fn next(&mut self) -> Option<Box<Cert>> {
+            self.0.pop().map(|cert| Box::new(Cert(cert)))
+        }
+    }
+
+    #[diplomat::opaque]
+    pub struct EncapsulatedContentInfo(pub picky_asn1_x509::content_info::EncapsulatedContentInfo);
+
+    impl EncapsulatedContentInfo {
+        pub fn content_type(&self, writable: &mut DiplomatWriteable) -> Result<(), Box<PickyError>> {
+            let oid: String = self.0.content_type.0.clone().into();
+            write!(writable, "{}", oid)?;
+            Ok(())
+        }
+    }
+
+    /// TODO: expose the SignerInfo struct public fields
+    #[diplomat::opaque]
+    pub struct SignerInfo(pub picky_asn1_x509::signer_info::SignerInfo);
+
+    #[diplomat::opaque]
+    pub struct SignerInfos(pub Vec<SignerInfo>);
+
+    impl SignerInfos {
+        pub fn next(&mut self) -> Option<Box<SignerInfo>> {
+            self.0.pop().map(Box::new)
+        }
+    }
```

Comment:
suggestion: Similar, you can call that `SignerInfoIterator` (no plural in this case)

Reply:
updated